### PR TITLE
Feat(docs): Docs for Operator Whitelisting with Multisig

### LINKED ISSUE #1493: docs: add whitelist process with multisig

### PR DESCRIPTION
**Pull request by PatStiles**
# Feat Docs for Operator Whitelisting with Multisig

## Description

Closes #1493 Docs for Operator Whitelisting with Multisig

## Type of change

Please delete options that are not relevant.

- [x] New feature

### CODE REVIEW

File: docs/0_internal/5_b_1_propose_whitelist.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+1. Go to [Safe](https://app.safe.global/home)
+
+2. Click on `New transaction` -> `Transaction Builder`
+
+   ![New transaction](./images/5_b_1_whitelist_operator_1.png)
+
+   ![Transaction Builder](./images/5_b_1_whitelist_operator_2.png)
+
+3. Get the `registryCoordinator` address from ```contracts/script/output/mainnet/alignedlayer_deployment_output.json``` or ```contracts/script/output/holesky/alignedlayer_deployment_output.json``` or ```contracts/script/output/sepolia/alignedlayer_deployment_output.json```
```
Comment (avilagaston9) [inline docs/0_internal/5_b_1_propose_whitelist.md:21]:
```suggestion
3. Get the `RegistryCoordinator` address from ```contracts/script/output/mainnet/alignedlayer_deployment_output.json``` or ```contracts/script/output/holesky/alignedlayer_deployment_output.json``` or ```contracts/script/output/sepolia/alignedlayer_deployment_output.json```
```

File: docs/0_internal/5_b_1_propose_whitelist.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+4. Paste the `registryCoordinator` address on `Enter Address or ENS Name`
```
Comment (avilagaston9) [inline docs/0_internal/5_b_1_propose_whitelist.md:23]:
```suggestion
4. Paste the `RegistryCoordinator` address on `Enter Address or ENS Name`
```

File: docs/0_internal/5_b_1_propose_whitelist.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+   ![](./images/5_b_1_whitelist_operator_3.png)
+
+5. As this is a Proxy contract, choose `Use Implementation ABI`
+
+   ![Use Implementation ABI](./images/5_b_1_whitelist_operator_4.png)
+
+If `Use Implementation ABI`, did not show up you will need to submit the call via raw calldata. Consult this this alternative [guide](./5_b_1b_propose_whitelist_with_call_data.md)
```
Comment (avilagaston9) [inline docs/0_internal/5_b_1_propose_whitelist.md]:
```suggestion
If `Use Implementation ABI`, did not show up you will need to submit the call via raw calldata. Consult this alternative [guide](./5_b_1b_propose_whitelist_with_call_data.md)
```

File: docs/0_internal/5_b_1_propose_whitelist.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+   
+6. In `contract method selector` choose `add_multiple()` in the `_addresses(address[])` field, enter the operator addresses in the following format `[<OPERATOR_ADDRESS>, ..., <OPERATOR_ADDRESS>]` for example, `[0000000000000000000000000000000000000009, 0000000000000000000000000000000000000003]`
```
Comment (avilagaston9) [inline docs/0_internal/5_b_1_propose_whitelist.md]:
```suggestion
6. In `Contract Method Selector` choose `add_multiple()` in the `_addresses(address[])*` field, enter the operator addresses in the following format `[<OPERATOR_ADDRESS>, ..., <OPERATOR_ADDRESS>]` for example, `[0000000000000000000000000000000000000009, 0000000000000000000000000000000000000003]`
```

File: docs/0_internal/5_b_2_approve_whitelist.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Approve the Whitelist Transaction
+
+Once the transaction is proposed, the multisig owners must approve the transaction.
+
+## Approve the Whitelist for registryCoordinator
```
Comment (avilagaston9) [inline docs/0_internal/5_b_2_approve_whitelist.md:5]:
```suggestion
## Approve the Whitelist for RegistryCoordinator
```

File: docs/0_internal/5_b_2_approve_whitelist.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+5. Copy the ```Raw Data```, paste it in a text editor and verify it is the same value as the one you got in step 3.
+
+6. If the data is correct, click on the `Confirm` button.
+
+7. Simulate the transaction. If everything is correct, click on the `Sign` button.
+
+   ![Sign transaction](images/5_b_2_whitelist_operator_2.png)
+
+8. Once the transaction is executed, the operator will be whitelisted within the `registryCoordinator` contract.
```
Comment (avilagaston9) [inline docs/0_internal/5_b_2_approve_whitelist.md:37]:
```suggestion
8. Once the transaction is executed, the operator will be whitelisted within the `RegistryCoordinator` contract.
```

File: docs/0_internal/5_b_whitelist_operators_with_multisig.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+   ```
+
+- Install foundry
+
+    ```shell
+    make install_foundry
+    foundryup -v nightly-a428ba6ad8856611339a6319290aade3347d25d9
+    ```
+
+## Steps for Pausing
```
Comment (avilagaston9) [inline docs/0_internal/5_b_whitelist_operators_with_multisig.md]:
```suggestion
## Steps for Whitelisting
```

File: docs/0_internal/5_b_whitelist_operators_with_multisig.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+1. Propose the transaction with the multisig following the [Propose Whitelist Guide](./5_b_1_propose_whitelist.md).
+
+2. After the Whitelist is proposed, multisig participants can approve the whitelist transaction following the [Approve Whitelist Guide](./5_b_2_approve_whitelist.md).
+
+## Steps for Unpausing
+
+1. Propose the transaction with the multisig following the [Propose Remove Whitelist Guide](./5_b_3_propose_remove_whitelist.md).
+
+2. After the unpause is proposed, multisig participants can approve the remove following the [Approve Remove Whitelist Guide](./5_b_4_approve_remove_whitelist.md).
```
Comment (avilagaston9) [inline docs/0_internal/5_b_whitelist_operators_with_multisig.md]:
```suggestion
```

File: docs/0_internal/5_b_1_propose_whitelist.md (hunk 2 above)
Comment (PatStiles) [reply docs/0_internal/5_b_1_propose_whitelist.md:23]: In the contract deployment config `registryCoordinator` is not capitalized. I think making it true to the config would help dissuade confusion.
Screenshot 2024-11-29 at 14 23 52

File: docs/0_internal/5_b_2_approve_whitelist.md (hunk 5 above)
Comment (PatStiles) [reply docs/0_internal/5_b_2_approve_whitelist.md:5]: In the contract deployment config `registryCoordinator` is not capitalized. I think making it true to the config would help dissuade confusion.

File: docs/0_internal/5_b_2_approve_whitelist.md (hunk 6 above)
Comment (PatStiles) [reply docs/0_internal/5_b_2_approve_whitelist.md:37]: In the contract deployment config `registryCoordinator` is not capitalized. I think making it true to the config would help dissuade confusion.

File: docs/0_internal/5_b_1_propose_whitelist.md (hunk 1 above)
Comment (PatStiles) [reply docs/0_internal/5_b_1_propose_whitelist.md:21]: In the contract deployment config `registryCoordinator` is not capitalized. I think making it true to the config would help dissuade confusion.

File: docs/0_internal/5_a_whitelist_operators.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
+# Whitelist Operators
+This doc contains a guide on how to Whitelist Operators in Aligned.
+
+To run the make targets specified in this guide, you must first have the relevant following env vars under `contracts/scripts/.env`:
+```
+export RPC_URL=<rpc_url>
```
Comment (JulianVentura) [inline docs/0_internal/5_a_whitelist_operators.md]: Here, remove the export keyboards

File: docs/0_internal/5_b_whitelist_operators_with_multisig.md (hunk 8 above)
Comment (PatStiles) [reply docs/0_internal/5_b_whitelist_operators_with_multisig.md]: addressed!

File: docs/0_internal/5_b_whitelist_operators_with_multisig.md (hunk 7 above)
Comment (PatStiles) [reply docs/0_internal/5_b_whitelist_operators_with_multisig.md]: addressed!

File: docs/0_internal/5_b_1_propose_whitelist.md (hunk 4 above)
Comment (PatStiles) [reply docs/0_internal/5_b_1_propose_whitelist.md]: addressed!

File: docs/0_internal/5_b_1_propose_whitelist.md (hunk 3 above)
Comment (PatStiles) [reply docs/0_internal/5_b_1_propose_whitelist.md]: addressed!

File: docs/0_internal/5_a_whitelist_operators.md (hunk 9 above)
Comment (PatStiles) [reply docs/0_internal/5_a_whitelist_operators.md]: addressed!

**Review by JulianVentura — APPROVED** [review]
Followed all guides and it worked as expected. Great work!

**Review by MarcosNicolau — APPROVED** [review]
Everything worked as expected.

**Review by JuArce — APPROVED** [review]

**Pull request merged**